projects/core/src/i18n/i18next/i18next-translation.service.spec.ts

```diff
@@ -49,142 +68,156 @@ describe('I18nextTranslationService', () => {
-        let result;
-        service
-          .translate(testKey, testOptions)
-          .pipe(first())
-          .subscribe((x) => (result = x));
-        expect(result).toBe(nonBreakingSpace);
-      });
-    });
-
-    describe(', when key does NOT exist firstly, but it comes with loaded chunk,', () => {
+    describe(' when i18next is NOT initialized', () => {
       beforeEach(() => {
-        spyOn(i18next, 'exists').and.returnValues(false, true);
-        spyOn(i18next, 'loadNamespaces').and.callFake(((
-          _namespaces,
-          onChunkLoad
-        ) => onChunkLoad()) as any);
+        i18next.isInitialized = false;
+        spyOn(i18next, 'loadNamespaces');
+        spyOn(i18next, 'exists');
+        spyOn(i18next, 't');
       });
 
-      it('should emit result of i18next.t', () => {
-        spyOn(i18next, 't').and.returnValue('value');
+      it('should return and not call translate method', () => {
         let result;
         service
-          .translate(testKey, testOptions)
+          .translate(testKey)
           .pipe(first())
           .subscribe((x) => (result = x));
-        expect(i18next.t).toHaveBeenCalledWith(
-          'testChunk:testKey',
-          testOptions
-        );
-        expect(result).toBe('value');
+        expect(result).toBe(undefined);
+        expect(i18next.loadNamespaces).not.toHaveBeenCalled();
+        expect(i18next.exists).not.toHaveBeenCalled();
+        expect(i18next.t).not.toHaveBeenCalled();
       });
     });
 
-    describe(', when language changed,', () => {
-      it('should emit result of i18next.t in new language', () => {
-        let languageChangedCallback;
-        spyOn(i18next, 'off');
-        spyOn(i18next, 'on').and.callFake(
-          (_event, callback) => (languageChangedCallback = callback)
-        );
-        spyOn(i18next, 'exists').and.returnValue(true);
-        spyOn(i18next, 't').and.returnValues('value1', 'value2');
-
-        let result;
-        service
-          .translate(testKey, testOptions)
-          .pipe(take(2))
-          .subscribe((x) => (result = x));
-        expect(result).toBe('value1');
-
-        languageChangedCallback();
-        expect(result).toBe('value2');
-
-        expect(i18next.off).toHaveBeenCalledWith(
-          'languageChanged',
-          languageChangedCallback
-        );
+    testKeys.forEach((key) => {
+      describe(getDescription(key), () => {
+        const namespacedKeys = getNamespacedKeys(key);
+        beforeEach(() => {
+          i18next.isInitialized = true;
+        });
+
+        describe(', when key exists,', () => {
+          beforeEach(() => {
+            spyOn(i18next, 'exists').and.returnValue(true);
+          });
+
+          it('should emit result of i18next.t', () => {
+            spyOn(i18next, 't').and.returnValue('value');
+            let result;
+            service
+              .translate(key, testOptions)
+              .pipe(take(namespacedKeys.length))
```

Comment:
Do we really want to `take` more than 1 emission from this service?

I believe it should emit just once, regardless the size of the fallback keys array.